client/src/luigi-client.js

```diff
@@ -2,15 +2,33 @@ import { lifecycleManager } from './lifecycleManager';
 import { linkManager } from './linkManager';
 import { uxManager } from './uxManager';
 import { storageManager } from './storageManager';
+import { helpers } from './helpers'; 
 
 /**
  * @name LuigiClient
  * @private
  */
 class LuigiClient {
   constructor() {
-    if (window !== window.top && window.document.head.getAttribute('disable-luigi-history-handling') != 'true') {
-      history.pushState = history.replaceState.bind(history);
+    if (window !== window.top){
+      if(window.document.head.getAttribute('disable-luigi-history-handling') != 'true') {
+        history.pushState = history.replaceState.bind(history);
+      }
+      if (window.document.head.getAttribute('disable-luigi-runtime-error-handling') != 'true') {
+        window.addEventListener('error', (error)=>{
+          const msg = {
+            msg:'luigi-runtim-error-handling',
+            errorObj:{
+              filename: error.filename,
+              message: error.message,
+              lineno: error.lineno,
+              colno: error.colno,
+              error: error.error
+            }
+          }
```

Comment:
```suggestion
        window.addEventListener('error', ({ filename, message, lineno, colno, error })=>{
          const msg = {
            msg:'luigi-runtime-error-handling',
            errorObj:{ filename, message, lineno, colno, error }
            }
          }
```
Please try this way and see it's still working properly.

Reply:
Or this way is even cleaner. Please double-check if it works fine.
```suggestion
        window.addEventListener('error', ({ filename, message, lineno, colno, error }: errorObj)=>{
          const msg = {
            msg:'luigi-runtime-error-handling',
            errorObj
            }
          }
```